Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -302,13 +302,20 @@ public class ProgramBuilder {
         return randVar(ofType: type)
     }
 
+    /// Returns a random variable from the outer scope.
+    public func randVarFromOuterScope() -> Variable {
+        assert(hasVisibleVariables)
+        return randVarInternal(fromOuterScope: true)!
+    }
+
     /// Returns a random variable satisfying the given constraints or nil if none is found.
-    private func randVarInternal(_ selector: ((Variable) -> Bool)? = nil) -> Variable? {
+    private func randVarInternal(_ selector: ((Variable) -> Bool)? = nil, fromOuterScope: Bool = false) -> Variable? {
         var candidates = [Variable]()
+        let scopes = fromOuterScope && scopeAnalyzer.scopes.count > 1 ? scopeAnalyzer.scopes.dropLast() : scopeAnalyzer.scopes
```

Comment:
I think you can remove `&& scopeAnalyzer.scopes.count > 1` because "If the number of elements to drop exceeds the number of elements in the collection, the result is an empty subsequence." (From https://developer.apple.com/documentation/swift/array/1689751-droplast). So if the API is used incorrectly, then this function will either just crash or return nil (which is fine), but won't accidentally return variables from the inner scope (which is probably not fine, because it might go unnoticed).